plugins/modules/bootc_manage.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/python
+
+# Copyright (c) 2024, Ryan Cook <rcook@redhat.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt
+# or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: bootc_manage
+author:
+- Ryan Cook (@cooktheryan)
+short_description: Bootc Switch and Upgrade
+description:
+    - This is a bootc module that manages the switching and upgrading of bootc.
+options:
+    state:
+        description:
+            - 'Control to apply the latest image or switch the image.'
+            - 'This is a string value.'
+            - 'NOTE: This will not reboot the system.'
+            - 'Please use the reboot module to reboot the system.'
+        required: true
+        type: str
+        choices: ['switch', 'latest']
+    image:
+        description:
+            - 'The image to switch to.'
+            - 'This is required when state is ''switch''.'
+        required: false
+        type: str
+
+'''
+
+EXAMPLES = '''
+# Switch to a different image
+- name: Provide image to switch to a different image and retain the current running image
+  community.bootc_manage:
+    state: switch
+    image: "example.com/image:latest"
+
+# Apply updates of the current running image
+- name: Apply updates of the current running image
+  community.bootc_manage:
+    state: latest
+'''
+
+RETURN = '''
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = dict(
+        state=dict(type='str', required=True, choices=['switch', 'latest']),
+        image=dict(type='str', required=False)
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    state = module.params['state']
+    image = module.params['image']
+
+    if state == 'switch':
+        if not image:
+            module.fail_json(msg="Image is required when state is 'switch'")
+        command = 'bootc switch {} --retain'.format(image)
```

Comment:
Prefer the form:
```suggestion
        command = ['bootc', 'switch', image, '--retain']
```
so that `run_command()` refrain from performing insecure shell interpolations.